internal/topo/rule/ruleState.go

```diff
@@ -220,6 +248,63 @@ func (rs *RuleState) Start() error {
 	if rs.triggered == -1 {
 		return fmt.Errorf("rule %s is already deleted", rs.RuleId)
 	}
+	if rs.Rule.IsScheduleRule() {
+		return rs.startScheduleRule()
+	}
+	return rs.start()
+}
+
+// startScheduleRule will register the job in the backgroundCron to run.
+// Job will do following 2 things:
+// 1. start the rule in cron if else the job is already stopped
+// 2. after the rule started, start an extract goroutine to stop the rule after specific duration
+func (rs *RuleState) startScheduleRule() error {
+	if rs.cronState.isInSchedule {
+		return fmt.Errorf("rule %s is already in schedule", rs.RuleId)
+	}
+	d, err := time.ParseDuration(rs.Rule.Options.Duration)
+	if err != nil {
+		return err
+	}
+	var cronCtx context.Context
+	cronCtx, rs.cronState.cancel = context.WithCancel(context.Background())
+	entryID, err := backgroundCron.AddFunc(rs.Rule.Options.Cron, func() {
+		err := func() error {
+			rs.Lock()
+			defer rs.Unlock()
+			if rs.triggered != 0 {
```

Comment:
If user stopped schedule rule manually, `cronState.isInSchedule` will be false, and it will remove the rule from the scheduler.

For now trigger indicates the real status whether the rule is actually running or not, so when user click `start` for schedule rule, it will be remain 0

Reply:
But triggered is not synced with the actual running status.  Maybe you don't need to worry about the "real" status, just trigger a start event in schedule time and let the start function handle it.

Reply:
updated.